contracts/src/CrossChainApplications/examples/NativeTokenBridge/NativeTokenDestination.sol

```diff
@@ -213,27 +280,46 @@ contract NativeTokenDestination is TeleporterOwnerUpgradeable, INativeTokenDesti
 
         (address recipient, uint256 amount) = abi.decode(message, (address, uint256));
         require(recipient != address(0), "NativeTokenDestination: zero recipient address");
-        require(amount != 0, "NativeTokenDestination: zero transfer value");
+
```

Comment:
Unrelated to this PR (and definitely not a change to be in this PR), but when we port these contracts over to the `teleporter-token-bridge` repo, I'd suggest that we create a `_handleMint` private helper function here rather than having the logic directly in `_receiveTeleporterMessage`.

I think a good standard pattern for `_receiveTeleporterMessage` implementations is:
- authenticate sender
- decode message
- call private handle function for message type.

Reply:
Yep, I agree